frontend/src/shared/utils/loadScript.ts

```diff
@@ -0,0 +1,18 @@
+export const loadScript = (source: string): Promise<void> => {
+  const element = document.querySelector(`script[src="${source}"]`);
+
+  if (element) {
+    return Promise.resolve();
+  }
+
+  return new Promise((resolve) => {
+    const script = document.createElement('script');
+
+    script.async = true;
+    script.defer = true;
+    script.src = source;
+    script.onload = () => resolve();
+
```

Comment:
```suggestion
    script.onerror = (error) => reject(error);
```
script 로드에 실패했을 때 에러 처리가 필요해 보입니다.